gym_electric_motor/state_action_processors/dq_to_abc_action_processor.py

```diff
@@ -0,0 +1,147 @@
+        def wrapper(callable_):
+            for motor_type in motor_types:
+                cls._registry[motor_type] = callable_
+        return wrapper
+
+    @classmethod
+    def make(cls, motor_type, *args, **kwargs):
+        assert motor_type in cls._registry.keys(), f'Not supported motor_type {motor_type}.'
+        class_ = cls._registry[motor_type]
+        inst = class_(*args, **kwargs)
+        return inst
+
+    def __init__(self, angle_name, physical_system=None):
+        """
+        Args:
+            angle_name(string): Name of the state that defines the current angle of the AC-motor
+            physical_system(PhysicalSystem(optional)): The inner physical system.
+        """
+        self._angle = 0.0
+        self._angle_index = None
+        self._omega_index = None
+        self._state = None
+        self._angle_name = angle_name
+        self._angle_advance = 0.0
+
+        super().__init__(physical_system)
+
+    def set_physical_system(self, physical_system):
+        # Docstring of super class
+        assert isinstance(physical_system.electrical_motor, ps.ThreePhaseMotor), \
+            'The motor in the system has to derive from the ThreePhaseMotor to define transformations.'
+        super().set_physical_system(physical_system)
+        self._omega_index = physical_system.state_names.index('omega')
+        self._angle_index = physical_system.state_names.index(self._angle_name)
+        assert self._angle_name in physical_system.state_names, \
+            f'Angle {self._angle_name} not in the states of the physical system. ' \
+            f'Probably a flux observer is required.'
+
+        self._angle_advance = 0.5
+
+        # If dead time has been added to the system increase the angle advance by the amount of dead time steps
+        if hasattr(physical_system, 'dead_time'):
+            self._angle_advance += physical_system.dead_time
+
+        return self
+
+    def simulate(self, action):
+        raise NotImplementedError
+
+    def reset(self, **kwargs):
+        # Docstring of super class
+        normalized_state = self._physical_system.reset()
+        self._state = normalized_state * self._physical_system.limits
+        return normalized_state
+
+    def _advance_angle(self, state):
+        return state[self._angle_index] \
+            + self._angle_advance * self._physical_system.tau * state[self._omega_index]
+
+
+class _ClassicDqToAbcActionProcessor(DqToAbcActionProcessor):
+
+    @property
+    def action_space(self):
+        return gym.spaces.Box(-1, 1, shape=(2,), dtype=np.float64)
+
+    def simulate(self, action):
+        # Docstring of superclass
+        advanced_angle = self._advance_angle(self._state)
+        abc_action = self._transformation(action, advanced_angle)
+        normalized_state = self._physical_system.simulate(abc_action)
+        self._state = normalized_state * self._physical_system.limits
+        return normalized_state
```

Comment:
I think the output of the simulation routine should also not incorporate abc measurements if the dq preprocessor is used.